.github/workflows/windows-x64.yaml

```diff
@@ -0,0 +1,47 @@
+name: windows-x64
+
+on:
+  push:
+    branches:
+      - master
+
+  pull_request:
+    branches:
+      - master
+
+  workflow_dispatch:
+
+concurrency:
+  group: windows-x64-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  windows_x64:
+    name: Windows x64
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest]
+        build_type: [Release, Debug]
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure CMake
+        shell: bash
+        run: |
+          cmake --version
+
+          mkdir build
+          cd build
+          cmake -A x64 -D CMAKE_BUILD_TYPE=${{ matrix.build_type }} -DCMAKE_INSTALL_PREFIX=./install ..
```

Comment:
Please see
https://github.com/csukuangfj/openfst/actions/runs/8780936061/job/24091877962#step:3:121


> I think CMake files only generate x64 arch. 

Actually, no.

>  Use -A option to specify architecture.

Please refer to
https://k2-fsa.github.io/sherpa/onnx/install/windows.html#bit-windows-x86
for how we build sherpa-onnx for 32-bit windows on 64-bit windows using CMake.

I will switch to `Ninja`.

```
Generators

The following generators are available on this platform (* marks default):
* Visual Studio 17 2022        = Generates Visual Studio 2022 project files.
                                 Use -A option to specify architecture.
  Visual Studio 16 2019        = Generates Visual Studio 2019 project files.
                                 Use -A option to specify architecture.
  Visual Studio 15 2017 [arch] = Generates Visual Studio 2017 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 14 2015 [arch] = Generates Visual Studio 2015 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 12 2013 [arch] = Deprecated.  Generates Visual Studio 2013
                                 project files.  Optional [arch] can be
                                 "Win64" or "ARM".
  Visual Studio 9 2008 [arch]  = Deprecated.  Generates Visual Studio 2008
                                 project files.  Optional [arch] can be
                                 "Win64" or "IA64".
  Borland Makefiles            = Generates Borland makefiles.
  NMake Makefiles              = Generates NMake makefiles.
  NMake Makefiles JOM          = Generates JOM makefiles.
  MSYS Makefiles               = Generates MSYS makefiles.
  MinGW Makefiles              = Generates a make file for use with
                                 mingw32-make.
  Green Hills MULTI            = Generates Green Hills MULTI files
                                 (experimental, work-in-progress).
  Unix Makefiles               = Generates standard UNIX makefiles.
  Ninja                        = Generates build.ninja files.
  Ninja Multi-Config           = Generates build-<Config>.ninja files.
  Watcom WMake                 = Generates Watcom WMake makefiles.
  CodeBlocks - MinGW Makefiles = Generates CodeBlocks project files
                                 (deprecated).
  CodeBlocks - NMake Makefiles = Generates CodeBlocks project files
                                 (deprecated).
  CodeBlocks - NMake Makefiles JOM
                               = Generates CodeBlocks project files
                                 (deprecated).
  CodeBlocks - Ninja           = Generates CodeBlocks project files
                                 (deprecated).
  CodeBlocks - Unix Makefiles  = Generates CodeBlocks project files
                                 (deprecated).
  CodeLite - MinGW Makefiles   = Generates CodeLite project files
                                 (deprecated).
  CodeLite - NMake Makefiles   = Generates CodeLite project files
                                 (deprecated).
  CodeLite - Ninja             = Generates CodeLite project files
                                 (deprecated).
  CodeLite - Unix Makefiles    = Generates CodeLite project files
                                 (deprecated).
  Eclipse CDT4 - NMake Makefiles
                               = Generates Eclipse CDT 4.0 project files
                                 (deprecated).
  Eclipse CDT4 - MinGW Makefiles
                               = Generates Eclipse CDT 4.0 project files
                                 (deprecated).
  Eclipse CDT4 - Ninja         = Generates Eclipse CDT 4.0 project files
                                 (deprecated).
  Eclipse CDT4 - Unix Makefiles= Generates Eclipse CDT 4.0 project files
                                 (deprecated).
  Kate - MinGW Makefiles       = Generates Kate project files (deprecated).
  Kate - NMake Makefiles       = Generates Kate project files (deprecated).
  Kate - Ninja                 = Generates Kate project files (deprecated).
  Kate - Ninja Multi-Config    = Generates Kate project files (deprecated).
  Kate - Unix Makefiles        = Generates Kate project files (deprecated).
  Sublime Text 2 - MinGW Makefiles
                               = Generates Sublime Text 2 project files
                                 (deprecated).
  Sublime Text 2 - NMake Makefiles
                               = Generates Sublime Text 2 project files
                                 (deprecated).
  Sublime Text 2 - Ninja       = Generates Sublime Text 2 project files
                                 (deprecated).
  Sublime Text 2 - Unix Makefiles
                               = Generates Sublime Text 2 project files
                                 (deprecated).
```

Reply:
```
CMake Error at CMakeLists.txt:2 (project):
  Generator
    Ninja
  does not support platform specification, but platform
    x64
  was specified.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
```

Ninja does not support specifying the arch to build. 

I am going to extend the build script to windows 32 so I will not switch to Ninja to keep the code consistent both for
windows64 and windows32.

Reply:
That was what I meant: The _concept_ of a Platform exists in a VS project file, so that `-A` is specific to this generator. This doesn't mean that Ninja _doesn't support_ 32-bit builds (you can build 32-bit executables for 64-bit Linux, too), it just means Ninja isn't built with that platform concept in mind. In cross-compilation (and compiling a 32-bit binary on a 64-bit platform _is_ a kind of cross-compilation), Linux-style equivalent of the MS-specific "Platform" is the [architecture triple](https://clang.llvm.org/docs/CrossCompilation.html#target-triple). And while CMake [indeed supports cross-compilation](https://cmake.org/cmake/help/book/mastering-cmake/chapter/Cross%20Compiling%20With%20CMake.html), it requires programming it into CMakeFiles. We better avoid that.

The [simpler approach is to set up the environment](https://discourse.cmake.org/t/how-to-build-32-bit-windows-apps-with-ninja/10070) following the VS command-line toolset setup routine. VS installs shortcuts that run `cmd` and make it source an environment setup script.

<img src="https://github.com/kkm000/openfst/assets/8228229/4db1d6e3-86f1-47a7-866b-8dc11576133c" width="240">

All these use the same script, only its argument is different. The issue (or non-issue) is locating this environment setup script _robustly_ is tricky: `cmd` is a weird language, `pwsh` is easier, but still takes some contortions. But there is an easy hack to _change_ the environment from one type to another, since the path to the script is already in an environment variable. I'm assuming you are using `cmd` as the shell; it simply won't work in WSL bash. 

1. Figure out the toolchain's host arch: run `echo %VSCMD_ARG_HOST_ARCH%` in the runner to see if it's `x64` (64-bit) or `x86` (32-bit). This is to preserve the host arch at step 3. I can't imagine that this value could ever change in the runner image, so we can simply hard-code it.
2. Reset a few environment variables that the setup script blindly augments:
  ```
set EXTERNAL_INCLUDE=
set INCLUDE=
set LIB=
set LIBPATH=
set PATH=%__VSCMD_PREINIT_PATH%
  ```
3. Find out the correct argument to the setup script:
 a. If the host arch is `x64`, the argument is `x64` to build 64-bit binaries, and `x64_x86` to cross-build 32-bit binaries.
 b. If the host arch is `x86`, the argument is `x86_x64` to cross-build 64-bit binaries, and `x86` to build 32-bit binaries.
4. Call the setup script with that argument (`call` to cmd is what's `source` to Bash):
  ```
call "%VCINSTALLDIR%\Auxiliary\Build\vcvarsall" <arg>
  ```

This is all to it. Further, you can even skip steps 1 and 3, and simply use `x64` or `x86` for the `<arg>` at step 4. This is the same as opening the shell set up with the "native" environment using the shortcuts in the picture for both 32 and 64 bit builds. The whole kaboodle is only to modify the environment as little as possible. Personally, I wouldn't care about preserving the host arch, and would do this, the simpler way.